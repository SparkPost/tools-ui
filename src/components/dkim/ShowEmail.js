import React from 'react';
import { LinkButton } from '../Button';

export default (props) => {
  const { email } = props;

  // TODO make email input copy-able with clipboard thing

  return (
    <div className='panel panel--accent text--left'>
      <div className='panel__body'>
        <h4>Validate your DKIM Signature</h4>
        <p className='paddingBottom--md'>Send an email to this generated email address, then view your results.</p>
        <div className='flex'>
          <div className='col-xs-8'>
            <input className='input__text input--full' type="text" readOnly={true} value={email} />
          </div>
          <div className='col-xs-4'>
            <LinkButton type='blue' fullWidth={true} to={`/dkim/results/${email}`}>View Results</LinkButton>
          </div>
        </div>
      </div>
    </div>
  );
};
