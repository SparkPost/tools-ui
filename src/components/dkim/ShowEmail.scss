@import "~@sparkpost/styles/src/assets/scss/shared/config";
@import "~@sparkpost/styles/src/assets/scss/shared/functions";

// Responsive box model helpers don't exist
// This is here only to produce a 1rem gap between the input and button on mobile
.showEmail {
  &__bump {
    margin-top: rem(18);

    @media screen and (min-width: break(md)) {
      margin-top: 0;
    }
  }
}
