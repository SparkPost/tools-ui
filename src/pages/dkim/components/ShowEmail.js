import React from 'react';
import { LinkButton } from 'components/button/Button';
import { CopyPopover } from 'components/popover/Popover';

export default (props) => {
  const { email } = props;

  return (
    <div className='panel panel--accent text--left'>
      <div className='panel__body'>
        <h4 className='marginBottom--xxs'>Validate your DKIM Signature</h4>
        <p className='paddingBottom--md'>Send an email to this generated email address, then view your results.</p>
        <div className='flex'>
          <div className='col-xs-12 col-md-8 col-lg-9'>
            <div className='input__group'>
              <input className='input__text input--full' type="text" readOnly={true} value={email} />
              <div className='input__buttonWrapper'>
                <CopyPopover placement='top' block={true} stringToCopy={email}>
                  <button className='button button--muted button--full'>Copy</button>
                </CopyPopover>
              </div>
            </div>
          </div>
          <div className='col-xs-12 col-md-4 col-lg-3 responsiveBump'>
            <LinkButton type='orange' fullWidth={true} to={`/dkim/results/${email}`}>View Results</LinkButton>
          </div>
        </div>
      </div>
    </div>
  );
};
