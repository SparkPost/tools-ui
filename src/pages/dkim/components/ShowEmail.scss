@import "~@sparkpost/styles/src/assets/scss/shared/config";
@import "~@sparkpost/styles/src/assets/scss/shared/functions";
