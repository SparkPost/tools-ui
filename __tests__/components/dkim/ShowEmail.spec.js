import React from 'react';
import renderer from 'react-test-renderer';

import ShowEmail from 'pages/dkim/components/ShowEmail';

describe('ShowEmail component', () => {

  test('should render', () => {
    expect(renderer.create(<ShowEmail email='some@email.com' />)).toMatchSnapshot();
  });

});
