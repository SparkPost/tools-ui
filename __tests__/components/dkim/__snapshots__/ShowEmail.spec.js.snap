exports[`ShowEmail component should render 1`] = `
<div
  className="panel panel--accent text--left">
  <div
    className="panel__body">
    <h4>
      Validate your DKIM Signature
    </h4>
    <p
      className="paddingBottom--md text--muted">
      Send an email to this generated email address, then view your results.
    </p>
    <div
      className="flex">
      <div
        className="col-xs-12 col-md-8 col-lg-9">
        <div
          className="input__group">
          <input
            className="input__text input--full"
            readOnly={true}
            type="text"
            value="some@email.com" />
          <div
            className="input__buttonWrapper">
            <span
              className="popover__group h-block"
              onClick={[Function]}>
              <button
                className="button button--muted button--full">
                Copy
              </button>
              <div
                className="popover copyPopover popover--l popover--rightAligned popover--top">
                <h6
                  className="text--left copyPopover__label">
                  <i
                    className="fa fa-check-circle" />
                   Copied to Clipboard!
                </h6>
                <input
                  className="input__text input__text--s copyPopover__input"
                  readOnly={true}
                  value="some@email.com" />
              </div>
            </span>
          </div>
        </div>
      </div>
      <div
        className="col-xs-12 col-md-4 col-lg-3 showEmail__bump">
        <a
          className="button button--orange button--full"
          onClick={[Function]}
          style={Object {}}>
          View Results
        </a>
      </div>
    </div>
  </div>
</div>
`;
